solutions/swb-app/src/datasetRoutes.ts

```diff
@@ -85,6 +87,26 @@ export function setUpDSRoutes(
     })
   );
 
+  // Delete dataset
+  router.delete(
+    '/datasets/:datasetId',
+    wrapAsync(async (req: Request, res: Response) => {
+      const { data: environments } = await metadataService.listDependentMetadata<Environment>(
```

Comment:
This doesn't actually return a full environment object right? It'll just return an array of `pk, sk, id`.  Typecasting the response to an `environment` is going to be misleading for developers since the IDE will show that the `environment` object has fields that it doesn't actually have.

Reply:
What if we define a Zod parser and type for `dataset with environment` metadata, and update the `listDependentMetadata` method to accept a Zod parser and return a Zod type? This could look like [this](https://github.com/aws-solutions/solution-spark-on-aws/blob/890744dd99dc857ebd83c56e20c5b7ac9eef3d4b/solutions/swb-app/src/costCenterRoutes.ts#L61). This will allow us to validate that the object type is correct.

Reply:
[This](https://github.com/aws-solutions/solution-spark-on-aws/blob/890744dd99dc857ebd83c56e20c5b7ac9eef3d4b/workbench-core/accounts/src/models/costCenters/updateCostCenterRequest.ts) is what a Zod parser and type looks like. In our case the Zod type will just have `id`, `pk`, `sk`, and `createdAt`.